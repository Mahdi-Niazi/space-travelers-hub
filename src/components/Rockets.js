import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Card from 'react-bootstrap/Card';
import { getRockets } from '../redux/slices/RocketsSlice';
import './module.Rockets.css';

const Rockets = () => {
  const { isLoading, isError, allRockets } = useSelector((store) => store.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getRockets());
  }, [dispatch]);

  const loading = isLoading && <p>Books loading, please wait!</p>;
  const error = isError && <p>Error loading books, please try again!</p>;

  return (
    <div>
      <div className="rockets-container">
        {loading}
        {error}
        {allRockets.map((rocket) => (
          <Card key={rocket.id} className="rocket-card">
            <Card.Img variant="top" src={rocket.flickr_images[0]} className="card-image" />
            <Card.Body>
              <Card.Title className="title">{rocket.rocket_name}</Card.Title>
              <Card.Text>{rocket.description}</Card.Text>
            </Card.Body>
          </Card>
        ))}
      </div>
    </div>
  );
};

export default Rockets;
