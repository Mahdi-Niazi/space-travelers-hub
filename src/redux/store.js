import { configureStore } from '@reduxjs/toolkit';

import missionReducer from './slices/MissionsSlice';

const store = configureStore({
  reducer: {
    missions: missionReducer,
  },
});

export default store;
