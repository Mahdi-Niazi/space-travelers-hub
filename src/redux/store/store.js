import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from '../slices/RocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsReducer,
  },
});

export default store;
