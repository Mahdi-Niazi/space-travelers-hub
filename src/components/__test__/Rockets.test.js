import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Rockets from '../Rockets';
import store from '../../redux/store';

describe('Rockets Component', () => {
  it('should match the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );

    expect(container).toMatchSnapshot();
  });
});
