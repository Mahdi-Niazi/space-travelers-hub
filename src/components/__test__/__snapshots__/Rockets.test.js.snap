// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets Component should match the snapshot 1`] = `
<div>
  <div>
    <div
      class="rockets-container"
    >
      <p>
        Rockets loading, please wait!
      </p>
    </div>
  </div>
</div>
`;
